language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Ck.hs

```diff
@@ -69,82 +72,83 @@ substituteDb varFor new = go where
 
 -- | The computing part of the CK machine. Rules are as follows:
 --
--- > s ▷ {M A}      ↦ s , {_ A} ▷ M
--- > s ▷ [M N]      ↦ s , [_ N] ▷ M
+-- > s ▷ {M A}      ↦ s , {_ A}        ▷ M
+-- > s ▷ [M N]      ↦ s , [_ N]        ▷ M
 -- > s ▷ wrap α A M ↦ s , (wrap α S _) ▷ M
--- > s ▷ unwrap M   ↦ s , (unwrap _) ▷ M
+-- > s ▷ unwrap M   ↦ s , (unwrap _)   ▷ M
 -- > s ▷ abs α K M  ↦ s ◁ abs α K M
 -- > s ▷ lam x A M  ↦ s ◁ lam x A M
 -- > s ▷ con cn     ↦ s ◁ con cn
 -- > s ▷ error A    ↦ ◆
-(|>) :: Context -> Term TyName Name () -> EvaluationResultDef
-stack |> TyInst _ fun ty        = FrameTyInstArg ty : stack |> fun
-stack |> Apply _ fun arg        = FrameApplyArg arg : stack |> fun
+(|>) :: Context -> Term TyName Name () -> CkM (Term TyName Name ())
+stack |> TyInst _ fun ty        = FrameTyInstArg ty      : stack |> fun
+stack |> Apply _ fun arg        = FrameApplyArg arg      : stack |> fun
 stack |> IWrap ann pat arg term = FrameIWrap ann pat arg : stack |> term
-stack |> Unwrap _ term          = FrameUnwrap : stack |> term
+stack |> Unwrap _ term          = FrameUnwrap            : stack |> term
 stack |> tyAbs@TyAbs{}          = stack <| tyAbs
 stack |> lamAbs@LamAbs{}        = stack <| lamAbs
 stack |> bi@Builtin{}           = stack <| bi
 stack |> constant@Constant{}    = stack <| constant
-_     |> Error{}                = EvaluationFailure
-_     |> var@Var{}              = throwCkMachineException OpenTermEvaluatedMachineError var
+_     |> Error{}                =
+    throwingWithCause _EvaluationError (UserEvaluationError ()) Nothing
```

Comment:
Why not the error term?

Reply:
For histerical reasons. Thanks, will fix.